src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -26,27 +27,115 @@ case class SimplePeriod(unit: TemporalUnit, n: Number, modifier: Modifier) exten
 case object UnknownPeriod extends Period
 case class PeriodSum(periods: Set[Period], modifier: Modifier) extends Period
 
-trait Interval extends Temporal
-case object DocumentCreationTime extends Interval
-case object UnknownInterval extends Interval
-case class Event(description: String) extends Interval
-case class Year(n: Int) extends Interval
-case class Decade(n: Int) extends Interval
-case class Century(n: Int) extends Interval
-case class TwoDigitYear(interval: Interval, twoDigits: Int) extends Interval
-case class ThisPeriod(interval: Interval, period: Period) extends Interval
-case class ThisRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class LastPeriod(interval: Interval, period: Period) extends Interval
-case class LastRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class NextPeriod(interval: Interval, period: Period) extends Interval
-case class NextRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class BeforePeriod(interval: Interval, period: Period) extends Interval
-case class BeforeRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class AfterPeriod(interval: Interval, period: Period) extends Interval
-case class AfterRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class Between(startInterval: Interval, endInterval: Interval) extends Interval
-case class Nth(interval: Interval, value: Int, repeatingInterval: RepeatingInterval) extends Interval
-case class IntervalSubIntervalIntersection(interval: Interval, subInterval: RepeatingInterval) extends Interval
+trait Interval extends Temporal {
```

Comment:
Add a little documentation for the start and end vars. In particular, you'll want to document that start is inclusive and end is exclusive (like you've done in the Intervals below).